### DISCUSSION TITLE
CSS Shortcuts

### DISCUSSION THREAD

**Discussion by mxmissile** [microsoft/fluentui-blazor#1431] [Q&A] [answered]
Does this lib have any css shortcuts or utilities similar to [bootstrap Spacing](https://getbootstrap.com/docs/5.3/utilities/spacing/)? Or [MudBlazor spacing](https://mudblazor.com/utilities/spacing#how-it-works)?

For example to quickly add padding you could use `pa-3` for padding all sides.

**Comment by vnbaaij** [comment, marked as answer]
No, we don't have anything like that and are not planning to add this.

**Comment by thgla** [comment]
I just updated a project from Bootstrap to Fluent UI, but kept the Bootstrap utilities because I'm used to them.
It's pretty easy to do: If you download the compiled & minified Bootstrap CSS files, there is a separate `bootstrap-utilities.min.css` you can just include in your project.

**Reply by StevenTCramer** [reply]
You can do similar with Tailwind.  Although more and more I'm just using <style> tags and vanilla css.

**Comment by lnaie** [comment]
That's not a great call, guys. We need a way to do style changes without loading yet another CSS library into the project.
With minimal effort, you could add some helpers, like spacing, margins, text sizes, etc.

**Reply by vnbaaij** [reply]
Why 'yet another'? By using our library, your are not using/loading a CSS library but a component package.

As mentioned by the others. there are multiple ways to add this to your project if you want/need this. It is not the goal of our package to provide this. Besides, I doubt if we would ever be able to do this in the ;right' way as everyone has their own preferences and (strong) opinions.

**Comment by lnaie** [comment]
Is it possible to have a serious component library offer without a default styling and theming system and its necessary helpers?
As it is, Blazor is already bloated. On wasm side, by the .net library size and its unfortunate load noise. If it's not that then it's the hardware resource limitation enforced by the SSR with web socket sessions.
It's hard to understand why the default strategy is to encourage using another css library that will certainly interfere with your theming and styling, create css bloat, and force devs to do extra work.

**Reply by dvoituron** [reply]
The next version (v5) will have Spacing, Margin and Padding utilities : https://fluentui-blazor-v5.azurewebsites.net/Spacing

**Reply by vnbaaij** [reply]
> I agree with you 100%.  This library has great potential but falls short in many areas inlcuding the spacing utilities.  Also, the collapsible menu is not a mobile firendly architecture.  Even in Bootstrap, I use Offscreen which can be acheived with Card & margin setup & all the acrobatics here but why should one waste time?  I am sticking with more thought-out libraries like MudBlazor, Blazorise etc.  Good luck folks.!

First of all, you are of course totally free to use whatever you want and what makes you happy. 

However, I don't recall you creating any issues or starting any discussions here about things you need to 'waste time on'. Nor have you done anything to improve this **open source** project by contributing fixes or improvements. Being critical is fine. Being critical and then leave with saying we fall short seems a bit easy to me...

**Reply by vnbaaij** [reply]
Denis (co-maintainer) already responded on what we are doing with this for the next major version. 

What I'm saying is that open source work best when it is a two way street. We build stuff (we think is useful), you suggest changes or additions or contribute fixes or whatever. Just trying it, not liking it and then say it falls short many areas without doing anything to try to improve it/make it better is not how it works.

**Comment by MBNSoftware** [comment]
Without getting into the debate, Blazorise handles this in a very nice way, in my opinion : https://blazorise.com/docs/helpers/utilities

**Reply by vnbaaij** [reply]
See the remark from Denis about the next version
